Minio/ApiEndpoints/BucketOperations.cs

```diff
@@ -201,62 +204,64 @@ public async Task SetVersioningAsync(SetVersioningArgs args, CancellationToken c
+
+#if NET2_0_OR_GREATER
+            var root = await XDocument.LoadAsync(responseResult.ContentStream, LoadOptions.None, cancellationToken).ConfigureAwait(false);
+#else
+            var root = XDocument.Load(responseResult.ContentStream);
+#endif
+
+            var items = from c in root.Root.Descendants(tag)
+                select new Item
                 {
-                    var goArgs = new GetObjectListArgs()
-                        .WithBucket(args.BucketName)
-                        .WithPrefix(args.Prefix)
-                        .WithDelimiter(delimiter)
-                        .WithVersions(args.Versions)
-                        .WithContinuationToken(nextContinuationToken)
-                        .WithMarker(marker)
-                        .WithListObjectsV1(!args.UseV2)
-                        .WithHeaders(args.Headers)
-                        .WithVersionIdMarker(versionIdMarker);
-                    if (args.Versions)
-                    {
-                        var objectList = await GetObjectVersionsListAsync(goArgs, cts.Token).ConfigureAwait(false);
-                        var listObjectsItemResponse = new ListObjectVersionResponse(args, objectList, obs);
-                        if (objectList.Item2.Count == 0 && count == 0) return;
-
-                        obs = listObjectsItemResponse.ItemObservable;
-                        marker = listObjectsItemResponse.NextKeyMarker;
-                        versionIdMarker = listObjectsItemResponse.NextVerMarker;
-                        isRunning = objectList.Item1.IsTruncated;
-                    }
-                    else
-                    {
-                        var objectList = await GetObjectListAsync(goArgs, cts.Token).ConfigureAwait(false);
-                        if (objectList.Item2.Count == 0 &&
-                            objectList.Item1.KeyCount.Equals("0", StringComparison.OrdinalIgnoreCase) && count == 0)
-                            return;
-
-                        var listObjectsItemResponse = new ListObjectsItemResponse(args, objectList, obs);
-                        marker = listObjectsItemResponse.NextMarker;
-                        isRunning = objectList.Item1.IsTruncated;
-                        nextContinuationToken = objectList.Item1.IsTruncated
-                            ? objectList.Item1.NextContinuationToken
-                            : string.Empty;
-                    }
-
-                    cts.Token.ThrowIfCancellationRequested();
-                    count++;
-                }
-            }
-        );
+                    Key = c.Element(ns + "Key").Value,
+                    LastModified = c.Element(ns + "LastModified").Value,
+                    ETag = c.Element(ns + "ETag").Value,
+                    Size = ulong.Parse(c.Element(ns + "Size").Value, CultureInfo.InvariantCulture),
+                    VersionId = c.Element(ns + "VersionId")?.Value ?? string.Empty,
```

Comment:
`VersionId` is only used when reading versions, so I set it to `string.Empty` instead if not present (could also be `null`).